src/Butler/App/TodoManager.hs

```diff
@@ -0,0 +1,149 @@
+                form_ [] $ do
+                    label_ [class_ "font-semibold"] "Task description"
+                    input_
+                        [ type_ "text"
+                        , placeholder_ "Enter task description"
+                        , name_ "taskDesc"
+                        , value_ ""
+                        , size_ "25"
+                        , maxlength_ "15"
+                        , class_ "h-8 ml-1 text-center border border-slate-300 rounded-md focus:border-slate-400"
+                        ]
+                    appSumitButton "Add Item"
+        div_ [] $ do
+            showItems ctx.wid appStateM
+
+showItems :: AppID -> MemoryVar TodoManager -> HtmlT STM ()
+showItems appID appStateM = do
+    (TodoManager _ todoTasks) <- lift $ readMemoryVar appStateM
+    div_ [class_ "flex flex-col m-2 gap-1"] $ do
+        forM_ todoTasks $ \(TodoTask taskId taskSelected taskDesc) -> do
+            div_ [class_ "flex flex-row gap-2"] $ do
+                withEvent appID "checkbox-click" [("taskID", Number $ fromInteger $ toInteger taskId)] $ do
+                    input_
+                        ( [type_ "checkbox"]
+                            <> case taskSelected of
+                                TaskSelected -> [checked_]
+                                TaskNotSelected -> mempty
+                        )
+
+                div_ [] $ toHtml taskDesc
+
+addTask :: TodoManager -> Text -> TodoManager
+addTask (TodoManager (TaskIndex i) todoTasks) content =
+    let newId = i + 1
+        task = newTask content (TaskId newId)
+     in TodoManager (TaskIndex newId) (task : todoTasks)
+
+delSelectedTasks :: TodoManager -> TodoManager
+delSelectedTasks (TodoManager taskIndex todoTasks) =
+    TodoManager taskIndex $
+        filter
+            ( \(TodoTask _ taskSelected _) ->
+                case taskSelected of
+                    TaskSelected -> False
+                    TaskNotSelected -> True
+            )
+            todoTasks
+
+selectTask :: TodoTask -> TodoTask
+selectTask (TodoTask taskId taskSelected taskDesc) =
+    let newSelectState = case taskSelected of
+            TaskSelected -> TaskNotSelected
+            TaskNotSelected -> TaskSelected
+     in TodoTask taskId newSelectState taskDesc
+
+newTask :: Text -> TaskId -> TodoTask
+newTask content taskId = TodoTask taskId TaskNotSelected $ TaskDesc content
+
+setSelectedTask :: TodoManager -> TaskId -> TodoManager
+setSelectedTask (TodoManager taskIndex todoTasks) taskId =
+    TodoManager taskIndex $
+        map
+            ( \task@(TodoTask tId _ _) ->
+                if tId == taskId
+                    then selectTask task
+                    else task
+            )
+            todoTasks
+
+startTodoManager :: AppContext -> ProcessIO ()
+startTodoManager ctx = do
+    logInfo "TodoManager started!" []
+    let appState = TodoManager 0 []
+        memAddr = "todo-manager-" <> showT ctx.wid <> ".bin"
```

Comment:
note that by using the `wid` for the storage address, if the app is closed and re-opened, then the memory is re-initialized. Though I don't know what would be the best solution. Perhaps we should use a single memoryvar, and handle synchronization when multiple todo app are running? Or we could use a service similar to the assistant, to dispatch updates when necessary? The chat app is using a broadcast chan to handle this workflow, but that's not ideal either, and it is missing persistence. It would be good to figure out a general purpose solution to render a common state to multiple app.

Reply:
Yes I saw that behavior. I tried to use a fixed name for the `memAddr` but then changes are not reflected between instances of the apps and probably storage will be overwritten between apps.

Either,
- the desktop does not allow multiple window of the same app
- or yes there is a wrapper like the assistant that provides the `memoryVar` capability and a way to advertise other instance of the same app that a change occurred. Could it be also like a transaction serial/number, this way an app can write in the memory by using the next serial ?

Reply:
Ha right, using a fixed `memAddr` wouldn't work. Incidentally, I've implemented a similar solution to share a common MemoryVar for the upcoming application setting feature, here is the gist of it: https://github.com/ButlerOS/haskell-butler/blob/34971485de9dc96dbb777fa6f6c239c122e5fe0f/src/Butler/AppSettings.hs#L38-L49 . Then to propagate change, we could use [renderOnChange](https://github.com/ButlerOS/haskell-butler/blob/bc5f1799b577bc4e6725bbf5dbf8b6a6be634f95/src/Butler/Display/GUI.hs#L45) like it is done for the file list: https://github.com/ButlerOS/haskell-butler/blob/bc5f1799b577bc4e6725bbf5dbf8b6a6be634f95/src/Butler/App/Desktop.hs#L51-L53

Reply:
Here is the proposed solution: https://github.com/ButlerOS/haskell-butler/pull/17